### DISCUSSION TITLE
DryIoc resolver tracing ?

### DISCUSSION THREAD

**Discussion by sigmarsson** [PrismLibrary/Prism#2691] [General] [answered]
Is there a way to trace the steps of `DryIoc` container resolver ? Currently [there is](https://stackoverflow.com/questions/72045760) a particular `(Uno.dll)` assembly could not be found while loading but unable to figure out why the assembly cannot be found on disk. I wonder if perhaps a `TraceListener` could be attached to the `PrismApplication` ?

**Comment by dansiegel** [comment]
In short Prism does not have a "Trace" mechanism on the Container. I'm not sure if DryIoc itself contains such functionality, perhaps @dadhi could answer that one.

**Comment by dadhi** [comment, marked as answer]
DryIoc itself does discover assembies, it just loads the Types from whatever is passed to it. 

Regarding Tracing it does not support it. But there are other things that may help with diagnostics. You may try to `Resolve<LambdaExpression>(yourType)` to examine the object graph. Or call `Validate(ServiceInfo.Of<YouType1>(), ...)`.

**Reply by dadhi** [reply]
If you want to know for sure if it DryIoc related you may call the DryIoc type loading on Uno.dll

```cs
public static Type[] GetLoadedTypes(this Assembly assembly)
        {
            try
            {
                return Portable.GetAssemblyTypes(assembly).ToArrayOrSelf();
            }
            catch (ReflectionTypeLoadException ex)
            {
                return ex.Types.Where(type => type != null).ToArray();
            }
        }
```

where `Portable.GetAssemblyTypes(assembly)`
is the plat. specific either `Assembly.DefinedTypes`  or `Assembly.GetTypes()`.
